dpypelines/s3_tar_received.py

```diff
@@ -1,22 +1,103 @@
+import importlib
+import json
+import os
+
+from dpytools.stores.directory.local import LocalDirectoryStore
+from dpypelines.pipeline.shared import notification as notify
+from dpypelines.pipeline.shared.schemas import get_config_schema_path
+from dpytools.s3.basic import decompress_s3_tar
+from dpytools.validation.json import validation
+from dpypelines.pipeline.shared import message
+
 def start(s3_object_name: str):
     """
     Handles the required behaviour when recieving a tar file indicated
     by an s3 object name.
 
-    Example s3_object_name: my-bucket/my-data.tar
+    Args:
+        s3_object_name (str): The S3 object name of the tar file to be processed.
+        Example: 'my-bucket/my-data.tar'
+
     """
 
-    # IN ALL CASES, the FOLLOWING STEPS WILL BE IN TRY CATCH
-    # BLOCKS AND WILL NOTIFY RELEVANT PARTIES IN EVENT OF FAILURE
+    # Decompress the tar file to the workspace
+    try:
+        decompress_s3_tar(s3_object_name, "input")
+    except Exception as err:
+        notify.data_engineering(message.unexpected_error(f"Failed to decompress tar file {s3_object_name}", err))
+        raise err
+    
+    # Identify the child directory that was just decompressed within /inputs
```

Comment:
so this - it turned out I was needing to do this because the tar file was created in an odd manner, so a folder was compressed to create it rather than just compressing some files into an archive at the root level.

so - we're just going to insist that there's no sub folder. The input files can then be decompressed directly into "./inputs".

so this can go, as can the reference to `{decompressed_directory}`.

it worked, but I don't want to get into handling sub folders, we'll just require that people provide things in the way we want (i.e not in a sub folder).